docs/spl-docs/evaluation-functions/text-functions.md

```diff
@@ -0,0 +1,186 @@
+- One can use this function with eval and where commands.
+
+#### Example
+The following command returns the left trimmed version (without any leading spaces or tabs) of the address.
+```
+... | eval left_trimmed_address=ltrim(address)
+```
+
+The following command trims the characters _a_, _Z_ and _space_ from the left side of the given string and returns this new string `"BcDeZ "`.
+```
+... | eval left_trimmed_str=ltrim("  aaBcDeZ ", "aZ ")
+```
+
+
+## rtrim(\<str\>, \<trim_chars\>)
+This function trims the characters present in the `<trim_chars>` from the right side of the `<str>` and returns this new string.
+
+`<trim_chars>` is an optional argument. If not present, it would remove leading spaces and tabs from the `<str>` and return this new string. Other types of whitespace characters would remain unaffected.
+
+#### Usage
+- `<str>` argument can be either a string literal or a field name. 
+- `<trim_chars>` is a string containing characters that needs to be trimmed. 
+- One can use this function with eval and where commands.
+
+#### Example
+The following command returns the right trimmed version (without any trailing spaces or tabs) of the address.
+```
+... | eval right_trimmed_address=rtrim(address)
+```
+
+The following command trims the characters _a_, _Z_ and space from the right side of the given string and returns this new string `"  aaBcDe"`.
+```
+... | eval right_trimmed_str=rtrim("  aaBcDeZ ", "aZ ")
+```
+
+
+## replace(\<str\>,\<regex\>,\<replacement\>)
+This function substitutes every occurence of the regular expression `<regex>` match in `<str>` with the `<replacement>` string.
+
+#### Usage
+- `<str>` argument can be either a string literal or a field name. 
+- `<regex>` is a string containing regular expression pattern. 
+- `<replacement>` is a string literal.
+- One can use this function with eval and where commands.
+
+#### Example
+The following command replaces the prefix of the string in field email with `"xxxxx"` and returns this new string which is a masked email.
+```
+... | eval masked_email=replace(email, "^([^@]+)@", "xxxxx@")
+```
+
+
+
+## substr(\<str\>,\<start\>,\<length\>)
+
+This functions returns a substring of `<str>` starting at index `<start>`. `<length>` denotes the number of characters to return from `<start>` index.
+
+#### Usage
+- `<str>` argument can be a string literal or a field name. 
+- `<length>` argument is optional if not present would return the rest of the string.
+- Indexes start at 1 not 0. Negative indexes can be used to indicate start from the end of the string.
+- One can use this function with eval and where commands.
+
+#### Example
+The following command returns the first 4 characters of string in field name.
+```
+... | eval substr_name=substr(name, 1, 4)
+```
+The following command returns the last -5 characters of the given string i.e. `"melon"`.
+```
+... | eval substr_str=substr("Watermelon", -5)
+```
```

Comment:
> returns the last -5 characters

Should be:
```
returns the last 5 characters
```

Reply:
Also, can you think of an example use case of `substr` that would solve an actual problem? That would be good to add here. Maybe extracting part of a message or specific field, or maybe extracting an API call giving the full url, or using substr for something else

In Mani's PR https://github.com/siglens/siglens-docs/pull/93/files#diff-315eff242bade8d33dde1d69e290fdcbb3590ef97ef510db79329316b9b52da1R13-R43 he has a really nice structure for giving examples. It would be good to follow that structure so that all of our SPL documentation has similar structure, which could help the reader understand the documentation just a little bit easier